File: algorithms/Middle of the Linked List/Middle of the Linked List.java
class Solution {
    public ListNode middleNode(ListNode head) {
        
        ListNode temp = head;
        int size = 0;
        while(temp!=null){
            size++;
            temp = temp.next;
        }
        int mid = size/2;
        temp = head;
        for(int i=0;i<mid;i++){
            temp = temp.next;
        }
        
        return temp;
        
    }
}

File: algorithms/M/Middle of the Linked List/Middle of the Linked List.py
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # basically we create two pointers
        # move one pointer extra fast
        # another pointer would be slow
        # when fast reaches end slow would be in mid
        slow = fast = head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
        return slow
